### DISCUSSION TITLE
Something similar for plugins?

### DISCUSSION THREAD

**Discussion by erdmann040** [wprig/wprig#796] [Q&A] [answered]
I really missed wprig and its integrated features when I had to develop a plugin. 

Can anyone recommend me a good basic plugin that has a similar philosophy as wprig and comes with a CSS preprocessor (it doesn't have to be PostCSS), Babel, JS compression etc. out of the box without much cofiguration on my side? 

Thanks in advance

**Comment by hsehszroc** [comment, marked as answer]
Could you look at this repo for the features you have mentioned: [Create WordPress Project](https://github.com/TheWebSolver/create-wordpress-project)

**Reply by erdmann040** [reply]
Thanks a lot!  Will try this out! Looks great :)

**Reply by robruiz** [reply]
@erdmann040 - How did this work out for you for a plugin? I'm curious.

**Comment by robruiz** [comment]
CSS in a plugin? I mean, I've done it, but...ok. I know that
the @wordpress/create-block package comes with webpack. So if you are
building a block, you get all of that setup out of the box. I usually start
plugins from the WPBP project, but it doesn't have the CSS build stuff
built-in. It's an interesting request, and to be honest, I have been
thinking about this quite a bit. I'm not sure why WP Rig needs to be only
for theme development. I end up developing more plugins than themes these
days, especially as things start to trend block-based. I'd be interested in
discussing this further. Are you in the WP Rig slack? If not, you should
get in there so we can begin a discussion there.

On Wed, Apr 20, 2022 at 1:51 AM Shesh Ghimire ***@***.***>
wrote:

> Could you look at this repo for the features you have mentioned: Create
> WordPress Project
> <https://github.com/TheWebSolver/create-wordpress-project>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wprig/wprig/discussions/796#discussioncomment-2598677>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB6LMGKXVQ2ZMXHCLW5J7JDVF6SPFANCNFSM5T2TTSIQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Rob Ruiz
Owner of Bizzle Designs
Web Design/Graphic Design/Sound Design/Audio & Video Production
www.bizzledesigns.com

**Reply by erdmann040** [reply]
I have to admit that I'm still struggling with blocks and React, but sooner or later I'll have to deal with it. For now, I'm mostly use shortcodes in the hopes that they will turn into blocks later. 

What exactly do you mean by wprig slick and how do I get into it? 

WP Rig is a bit like my dishwasher: in the beginning I had to invest a lot, but now I don't know how I could ever lived without it. It was definitely worth the investment. :)

If I could also use it for plugin development (or if there was a similar sister project), that would make WP Rig even more attractive. Sometimes I've integrated features that would have been better placed in a plugin into the theme just to get all the nice features of WP Rig. 

It would be great if I could use WP Rig for plugin development as well, but I didn't want to ask for it because I thought you and the team must have enough to do already.

**Comment by robruiz** [comment]
Use this to join our Slack:
https://join.slack.com/t/wprig/shared_invite/zt-162vhist3-XmeuQPLarTqw3~GrIlwIkA

Learning Webpack would be good for your needs, but it's not too easy to
understand. I will let you know if I figure anything out for what you are
trying to do. Perhaps there is an easy enough tutorial out there for how to
easily add Webpack to a WP plugin to build SASS or something like that.

On Wed, Apr 20, 2022 at 5:56 PM Jan Tack ***@***.***> wrote:

> I have to admit that I'm still struggling with blocks and React, but
> sooner or later I'll have to deal with it. For now, I'm mostly use
> shortcodes in the hopes that they will turn into blocks later.
>
> What exactly do you mean by wprig slick and how do I get into it?
>
> WP-Rig is a bit like my dishwasher: in the beginning I had to invest a
> lot, but now I don't know how I ever lived without it. It was definitely
> worth the investment.
>
> If I could also use it for plugin development (or if there was a similar
> sister project), that would make WP-Rig even more attractive. Sometimes
> I've integrated features that would have been better off in a plugin into
> the theme to get all the nice features of Wp-Rig.
>
> It would be great if you could use Wp-Rig for plugin development as well,
> but I didn't want to ask for it because I thought you and the team must
> have enough to do already.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wprig/wprig/discussions/796#discussioncomment-2605009>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB6LMGNY5PRC2ABBGXGVXB3VGCDRDANCNFSM5T2TTSIQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Rob Ruiz
Owner of Bizzle Designs
Web Design/Graphic Design/Sound Design/Audio & Video Production
www.bizzledesigns.com

**Reply by erdmann040** [reply]
Thank you so much for your willingness to help. I've already started looking into webpack and found a course on LinkedIn Learning that seems pretty good. I'll see if that helps me out. If not I'll get back to you.

**Reply by erdmann040** [reply]
You are probably right that a lot of CSS code in a plugin is a bad smell. Transpiling, uglyfying and compressing the JS code is certainly the higher priority.
